### PR TITLE
Complete initial reference documentation

### PR DESCRIPTION
**Pull request by ntadej**
Complete initial reference documentation:
- add widgets documentation
- add location documentation -> this is done manually using dummy placeholder QML files and `doxyqml`
- make sure include headers are hinted for all public C++ classes

### CODE REVIEW

File: .github/workflows/docs-test.yml [hunk 1]
```diff
@@ -33,6 +33,9 @@ jobs:
         with:
           version: "1.10.0"
 
+      - name: Install doxyqml
```
Comment (HarelM) [inline .github/workflows/docs-test.yml:36]: Consider using a docker image instead (if it exists).

File: docs/examples/example_Style.qml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2023 MapLibre contributors
```
Comment (HarelM) [inline docs/examples/example_Style.qml:1]: 2023?

**Review by HarelM — APPROVED** [review]
Looks very manual, I would consider automating some of the things here somehow...

File: .github/workflows/docs-test.yml (hunk 1 above)
Comment (ntadej) [reply .github/workflows/docs-test.yml:36]: Not that I am aware of. That's why I went with `pip install` as the recommended install path.

File: docs/examples/example_Style.qml (hunk 2 above)
Comment (ntadej) [reply docs/examples/example_Style.qml:1]: Indeed I need to update copyright headers in those files to match what we promise people. This repo is a bit messy as we inherited some old files from Mapbox/Qt.

**Comment by ntadej** [comment]
> Looks very manual, I would consider automating some of the things here somehow...

Quick investigation did not yield any automated tools. There is `QDoc` provided by Qt, but I had very hard time to get anything useful out of it. For now I think this is sensible enough, but I'll keep investigating if something (semi)automated is possible.

**Review by birkskyum — APPROVED** [review]

**Pull request merged**